runtime/gc_vlhgc/IncrementalGenerationalGC.cpp

```diff
@@ -932,10 +1017,28 @@ MM_IncrementalGenerationalGC::runPartialGarbageCollect(MM_EnvironmentVLHGC *env,
 	 * Collection end work
 	 */
 
-
 	postCollect(env, env->_cycleState->_activeSubSpace);
 }
 
+void
+MM_IncrementalGenerationalGC::runPartialGarbageCollect(MM_EnvironmentVLHGC *env, MM_AllocateDescription *allocDescription)
+{
+	bool isConcurrentCopyForwardEnabled = _extensions->isConcurrentCopyForwardEnabled();
+	/* If concurrent Copy Forward is enabled we only want to pre process at first PGC increment
+	 * and isConcurrentCycleInProgress() tells us if this is the first PGC increment or not */
+	if (!isConcurrentCopyForwardEnabled || !_copyForwardDelegate.isConcurrentCycleInProgress()) {
```

Comment:
Check against isConcurrentCopyForwardEnabled (and below for PostWrok) seems to be redundant, afterall (if concurrent is disabled, isConcurrentCycleInProgress is always false, making this expression true regardless).

Instead, you can assert that inside isConcurrentCycleInProgress(), in case it returns true, we could assert isConcurrentCopyForwardEnabled is true, too (CFDelegate seems to have access to GCExt).

Small differences, very subjective - up to you if you want to change anything about it.

Reply:
That makes sense and it would look cleaner

Reply:
It might be best for us to just remove `isConcurrentCopyForwardEnabled` and not bother with the assertion and keep `isConcurrentCycleInProgress()` simple, similar to its Scavenger counterpart.